dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/message/MessageImpl.java

```diff
@@ -184,7 +328,218 @@ public void setInterceptorChain(InterceptorChain ic) {
     
     @Override
     public Object getContextualProperty(String key) {
-        Object o = getOrDefault(key, NOT_FOUND);
+        //Liberty code change start
+        Object o = null;
```

Comment:
This is where if the KEYS was a HashMap of keys to Integers, if it returned non null you could just update this complex logic with:
Integer index = KEYS.get(key);
if (index != null) {
    Object value = propertyValues[index];
    return value == NOT_FOUND ? null : value;
}

Otherwise right now if it is NOT_FOUND we still look the old way.  Not sure why we should do that.  If it isn't set, we should just return null right?

The other major flaw here is that we are doing == checks on the key.  So if you are going to do == check on the you would would have to flow down to the old way of doing things because non == versions of the key would be in the Map.  This all causes problems with consistency though and can't be done like that.  If you want an example, let me know.